### PR TITLE
[hail] Remove Type.physicalType (!)

### PR DESCRIPTION
**Pull request by tpoterba**
Stacked on #7868

### CODE REVIEW

**Comment by akotlar** [comment]
Conflict to fix. Will review more tomorrow

**Comment by tpoterba** [comment]
oops, meant to apply stacked tag. stacked on the parser one.

**Comment by akotlar** [comment]
Yep, just got to the parser position :)

**Comment by tpoterba** [comment]
ready for review

File: hail/src/main/scala/is/hail/expr/types/physical/PType.scala [hunk 1]
```diff
@@ -276,7 +276,7 @@ abstract class PType extends Serializable with Requiredness {
 
   def subsetTo(t: Type): PType = {
     // FIXME
-    t.physicalType
+    PType.canonical(t)
```
Comment (akotlar) [inline hail/src/main/scala/is/hail/expr/types/physical/PType.scala]: I think this is stacked on the subsetTo pr

Comment (akotlar) [reply hail/src/main/scala/is/hail/expr/types/physical/PType.scala]: (And since that has a more complete solution, we can omit this change)

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/types/physical/PType.scala]: not stacked -- they'll just conflict. Whichever goes in first will have to change the other.

**Comment by akotlar** [comment]
merge needed

**Comment by akotlar** [comment]
needs bump

**Pull request merged**